### PR TITLE
Bartercast statistics community

### PR DESCRIPTION
**Pull request by corpaul**
Adds bartercast4 community and statistics for tunnel and torrent received stats. Also adds statistics database and debug panel for these stats. (works with dispersy/devel)

### CODE REVIEW

**Comment by whirm** [comment]
retest this please

**Comment by corpaul** [comment]
retest this please

**Comment by corpaul** [comment]
retest this please

**Comment by corpaul** [comment]
retest this please

**Comment by synctext** [comment]
@whirm can you see what is going wrong here?

**Comment by whirm** [comment]
retest this please

File: Tribler/community/bartercast4/community.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+                yield DelayMessageByProof(message)
+
+    def on_stats_request(self, messages):
+        log.msg("IN: stats-request")
+        for message in messages:
+            log.msg("stats-request: %s %s" % (message._distribution.global_time, message.payload.stats_type))
+            # send back stats-response
+            self.create_stats_response(message.payload.stats_type, message.candidate)
+
+    # todo
```
Comment (whirm) [inline Tribler/community/bartercast4/community.py]: can this go?

**Comment by whirm** [comment]
@corpaul I think it would be nice to have some tests for this. Do you think you will have time to do it before next week?
@synctext: If @corpaul can't make it. Should we merge this without having tests for it so the students can work on it or?

**Comment by synctext** [comment]
@whirm if it passes and you find the coding approach acceptable, it can be merged. Please bug Corpaul or create ticket for a test of this.

**Pull request merged**